lightkurve/lightcurve.py

```diff
@@ -1087,3 +1089,110 @@ def opt_depth(to, width, data):
         snr_d.append(depth_star * np.sqrt(width_star))
 
     return log_posterior, trial_periods, trial_periods[np.argmax(log_posterior)]
+
+class LightCurveCollection(object):
+    """
+    Collects multiple LightCurve objects together with helpful functions.
+    """
+    def __init__(self, lcs):
+        try:
+            lightcurves = np.asarray(lcs)
+        except:
+            raise TypeError("Unable to parse input")
+        self.data = {}
+        
+        for lc in lightcurves:
+            if isinstance(lc, KeplerLightCurve) or isinstance(lc,LightCurve):
+                if lc.keplerid:
+                    if lc.keplerid in self.data:
+                        self.data[lc.keplerid].append(lc)
+                    else:
+                        self.data[lc.keplerid] = [lc]
+                else: #no keplerID
+                    self.data[None].append(lc)
+            else:
+                raise TypeError("Object is not a LightCurve instance")
+        self.data[None] = []
+
+    def __len__(self):
+        length = 0
+        for lc_array in self.data.values():
+            for lc in lc_array:
+                length += 1
+        return length
+
+    def _ids(self):
+        """
+        Returns the kepler_ids of all the lightcurves as a dict_keys obj.
+        """
+        return self.data.keys()
+
+    def __getitem__(self, kep_id):
+        """
+        Returns the lightcurve associated with the kepler_id. 
+        """
+        try: 
+            return self.data[kep_id]
+        except:
+            raise ValueError('No LightCurve for ' + kep_id)
+
+    def append(self, lc, **kwargs):
+        try:
+            if lc.keplerid in self.data:
+                if 'stitch' in kwargs:
+                    self.data[lc.keplerid][0].append(lc)
+                    return   
+                self.data[lc.keplerid].append(lc)
+            else:
+                self.data[lc.keplerid] = [lc]
+        except:
+            raise TypeError("Input is not a lightcurve")
+
+    def stitch(self, normalize=False):
+        new_lc = LightCurve(time=[])
```

Comment:
Can you initialize a `LightCurve` with no flux now? This must be new, I'm surprised this doesn't fail. `LightCurve`'s should always have flux! It's fine here so long as it doesn't break but I would use
```
 new_lc = LightCurve(time=[], flux=[])
```
just in case.

Reply:
Done